libp2p/protocols/secure/noise.nim

```diff
@@ -124,13 +121,12 @@ proc encryptWithAd(state: var CipherState, ad, data: openarray[byte]): seq[byte]
   result &= tag
   trace "encryptWithAd", tag = byteutils.toHex(tag), data = result.shortLog, nonce = state.n - 1
 
-proc decryptWithAd(state: var CipherState, ad, data: openarray[byte]): seq[byte] =
+proc decryptWithAd(state: var CipherState, ad, data: openArray[byte]): seq[byte] =
   var
-    tagIn = data[^ChaChaPolyTag.len..data.high].intoChaChaPolyTag
-    tagOut = tagIn
+    tagIn = data.toOpenArray(data.len - ChaChaPolyTag.len, data.high).intoChaChaPolyTag
+    tagOut: ChaChaPolyTag
```

Comment:
where is `tagOut` being set now?

Reply:
`tagOut` comes from the encrypted data (it's appended as part of encryption and verified as part of decryption)

Reply:
`ChaChaPoly.decrypt` is a proc(..., var _: T, ...)

Reply:
Ugh, ok, but looks fishy, it either was wrong or it's wrong now. We used to set `tagOut = tagIn`, now we aren't...

Reply:
the underlying chacha it calls is `out-only` for tag

Reply:
I guess the chacha api could be made better but whatever